tasks/main.yml

```diff
@@ -17,26 +12,287 @@
+      service:
+        name: mssql-server
+        state: stopped
+      notify: Restart the mssql-server service
+      when: not __mssql_edition_matches.stdout | bool
+
+    - name: Change the edition of MSSQL
+      command: /opt/mssql/bin/mssql-conf set-edition
+      environment:
+        MSSQL_PID: "{{ mssql_edition }}"
+      register: __mssql_conf_set_edition
+      changed_when: '"The new edition is" in __mssql_conf_set_edition.stdout'
+      when: not __mssql_edition_matches.stdout | bool
+      notify: Restart the mssql-server service
+
+- name: Set the IP address setting for MSSQL
+  when: mssql_ip_address is not none
+  block:
+    - name: Check if the IP address setting is defined
+      command: /opt/mssql/bin/mssql-conf get network ipaddress
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_conf_get_ip_address
+
+    - name: Change the IP address setting
+      command: >
+        /opt/mssql/bin/mssql-conf set network.ipaddress {{ mssql_ip_address }}
+      when: >
+        ("No setting for the given" in __mssql_conf_get_ip_address.stdout) or
+        (__mssql_conf_get_ip_address.stdout | regex_replace( 'ipaddress : ', '')
+        not in mssql_ip_address)
+      notify: Restart the mssql-server service
+
+- name: Set the new TCP port for MSSQL
+  command: "/opt/mssql/bin/mssql-conf set network.tcpport {{ mssql_tcp_port }}"
+  register: __mssql_conf_set_tcp_port
+  ignore_errors: true
+  changed_when: >
+    "SQL Server needs to be restarted in order to apply this setting."
+    in __mssql_conf_set_tcp_port.stdout
+  notify: Restart the mssql-server service
+  when: mssql_tcp_port is not none
+
+- name: Verify that the mssql_password variable is defined
+  assert:
+    that:
+      - mssql_password is not none
+    fail_msg: >-
+      You must define the mssql_password variable because MSSQL requires
+      the sa user to authenticate to input SQL files.
+  when: mssql_input_sql_file is not none
+
+- name: Input the {{ mssql_input_sql_file }} file to MSSQL
+  when: mssql_input_sql_file is not none
+  block:
+    - name: Create a tempfile on the host
+      tempfile:
+        state: file
+      register: __mssql_sql_tempfile
+
+    - name: Copy the {{ mssql_input_sql_file }} file to the host
+      copy:
+        src: "{{ mssql_input_sql_file }}"
+        dest: "{{ __mssql_sql_tempfile.path }}"
+        mode: preserve
+
+    - name: Ensure that the mssql-server service is started
+      service:
+        name: mssql-server
+        state: started
+
+    - name: Prepare MSSQL and facts for logging in
+      include_tasks: verify_password.yml
+      vars:
+        __mssql_password: "{{ mssql_password }}"
+
+    - name: Input the {{ mssql_input_sql_file }} file with the sqlcmd command
+      command: >-
+        {{ __mssql_sqlcmd_login_cmd }} -i {{ __mssql_sql_tempfile.path }} -b
+      register: __mssql_sqlcmd_input_file
+      ignore_errors: true
```

Comment:
I think you should add a `changed_when: true` here to make this idempotent.
Note that this will essentially ignore any/all errors, making it the responsibility of the user to examine the output log for issues.  I guess this is ok until we can figure out how to do real error handling and idempotency.

Reply:
The command module always returns the Changed status unless `changed_when: false` is defined, isn't it? The `-b` flag that I added returns `rc=1` and this task fails if any error occurs, for the user to pay attention and observe the output. There is a task that prints the output of the input command too.

Reply:
ok - then let's leave this as-is, and see what feedback we get